lib/tasks/publishing_api.rake

```diff
@@ -84,4 +84,35 @@ namespace :publishing_api do
     )
     puts "Links patched for root page..."
   end
+
+  desc "Update redirect for an archived tag"
+  task :update_redirect, %i[content_id new_redirect_path] => :environment do |_task, args|
+    unless args[:new_redirect_path].present? && args[:new_redirect_path].starts_with?("/")
+      raise "new_redirect_path is missing or invalid"
+    end
+
+    if Services.content_store.content_item(args[:new_redirect_path])
+      tag = Tag.find_by(content_id: args[:content_id])
+
+      raise "No tag can be found for that content id" if tag.blank?
+
+      raise "This task can only be used for archived topics" unless tag.state == "archived" && tag.type == "Topic"
```

Comment:
Is it helpful to add the states and types to an enum?

Reply:
Unless you feel strongly, as this is a temporary task which I will remove shortly I would probably lean towards leaving as it is! Is that OK with you?